### ISSUE TITLE
Import its own theme

### ISSUE THREAD

**Issue by vincowl** [creecros/Customizer#66]
It would be great to be able to create its own theme and import it. Before 1.7.2, it was quite available, being forced to update Plugin.php file and copy your own css theme in Assets/css folder. Now, process is not documented, has changed. I managed to hack this but it would be great if it were a default behavior.

**Comment by creecros** [comment]
Again already addressed. The new folder will be located in the data folder, preview the css-generator branch for more details. I will re document the process upon merge.

**Comment by creecros** [comment]
Also, thanks for the input, stay tuned!

**Comment by creecros** [comment]
@vincowl 

> It would be great to be able to create its own theme and import it. Before 1.7.2, it was quite available, being forced to update Plugin.php file and copy your own css theme in Assets/css folder. Now, process is not documented, has changed. I managed to hack this but it would be great if it were a default behavior.

Also, to describe how it works now for version >1.7, you don't need to change anything in Plugin.php, simply drop your css file into the `Asset/css/themes' folder, the plugin takes care of the rest using a scan directory: `scandir('plugins/Customizer/Assets/css/themes`. It then builds and array based on the files in that folder.

```php
foreach ($scanned_themes as $theme) {
		$customizer['themes'][rtrim($theme, '.css')] = 'plugins/Customizer/Assets/css/themes/' . $theme;
	}
```

The new version which has not merged yet, will continue to do this, but will also scan the `DATA_DIR` in a similar place, for custom themes, that will not be destroyed when you upgrade versions. It will also use this folder for the **Theme Creator** function that is being added, so those themes will also remain after upgrades etc...

**Comment by creecros** [comment]
https://github.com/creecros/Customizer/releases/tag/1.8.0

**Comment by vincowl** [comment]
Great !! Could you just give me an advice of what DATA_DIR stands for ? No
trace of it found in the Docker installation env vars...
Also, 1.8.0 doesn't appear as an available upgrade in the Kanboard's plugin
manager. Is there something to set il or whatever ? Or am I forced to
install it manualy ?

Le jeu. 29 nov. 2018 à 21:09, Craig Crosby <notifications@github.com> a
écrit :

**Comment by creecros** [comment]
`DATA_DIR` is the constant var defined for your data folder. If you are using docker, it's one of the folders you mapped. i.e. `var/www/app/data`

1.8.0 was just released, it won't be available until the PR, kanboard/website#112 is merged.

You could install manually if you don't want to wait.

**Comment by creecros** [comment]
https://github.com/kanboard/kanboard/blob/cddaf2ac2ec541b695a7dcc2f7410322950c5eef/config.default.php#L10
